libs/Microsoft.MixedReality.WebRTC/Transceiver.cs

```diff
@@ -0,0 +1,299 @@
+        public string[] StreamIDs { get; }
+
+        /// <summary>
+        /// Backing field for <see cref="DesiredDirection"/>.
+        /// </summary>
+        /// <seealso cref="DesiredDirection"/>
+        protected Direction _desiredDirection;
+
+        /// <summary>
+        /// Handle to the native transceiver object, valid until <see cref="OnDestroyed"/>
+        /// is called by the native implementation when the transceiver is destroyed as part
+        /// of the peer connection closing.
+        /// </summary>
+        /// <remarks>
+        /// In native land this is a <code>mrsTransceiverHandle</code>.
+        /// </remarks>
+        internal IntPtr _nativeHandle = IntPtr.Zero;
+
+        /// <summary>
+        /// Reference to the struct keeping the callback delegates alive while registered with
+        /// the native implementation.
+        /// This should be released with <see cref="Utils.ReleaseWrapperRef(IntPtr)"/>.
+        /// </summary>
+        /// <seealso cref="TransceiverInterop.RegisterCallbacks(Transceiver, out IntPtr)"/>
+        private IntPtr _argsRef = IntPtr.Zero;
+
+        /// <summary>
+        /// Create a new transceiver associated with a given peer connection.
+        /// </summary>
+        /// <param name="handle">Handle to the native transceiver object.</param>
+        /// <param name="mediaKind">The media kind of the transceiver and its tracks.</param>
+        /// <param name="peerConnection">The peer connection owning this transceiver.</param>
+        /// <param name="mlineIndex">The transceiver media line index in SDP.</param>
+        /// <param name="name">The transceiver name.</param>
+        /// <param name="streamIDs">Collection of stream IDs the transceiver is associated with, as set by the peer which created it.</param>
+        /// <param name="initialDesiredDirection">Initial value to initialize <see cref="DesiredDirection"/> with.</param>
+        protected Transceiver(IntPtr handle, MediaKind mediaKind, PeerConnection peerConnection, int mlineIndex,
+            string name, string[] streamIDs, Direction initialDesiredDirection)
+        {
+            Debug.Assert(handle != IntPtr.Zero);
+            _nativeHandle = handle;
+            MediaKind = mediaKind;
+            PeerConnection = peerConnection;
+            MlineIndex = mlineIndex;
+            Name = name;
+            StreamIDs = streamIDs;
+            _desiredDirection = initialDesiredDirection;
+            TransceiverInterop.RegisterCallbacks(this, out _argsRef);
+        }
+
+        /// <summary>
+        /// Change the media flowing direction of the transceiver.
+        /// This triggers a renegotiation needed event to synchronize with the remote peer.
+        /// </summary>
+        /// <param name="newDirection">The new flowing direction.</param>
+        /// <seealso cref="DesiredDirection"/>
+        /// <seealso cref="NegotiatedDirection"/>
+        public void SetDirection(Direction newDirection)
```

Comment:
nit: IMHO a `SetDirection` and a settable `Direction` are redundant

Reply:
Fixed.